src/devices/Common/Iot/Device/Common/GeographicPosition.cs

```diff
@@ -0,0 +1,371 @@
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Returns true if the two positions are (almost) equal. This ignores the altitude.
+        /// </summary>
+        /// <param name="position">Position to compare with</param>
+        /// <returns>True if the two positions are closer than about 1cm</returns>
+        public bool EqualPosition(GeographicPosition position)
+        {
+            if (position == null)
+            {
+                return false;
+            }
+
+            bool ret;
+            if ((Math.Abs((position.Longitude - Longitude)) < ComparisonEpsilon) && (Math.Abs(position.Latitude - Latitude) < ComparisonEpsilon))
+            {
+                ret = true;
+            }
+            else
+            {
+                ret = false;
+            }
+
+            return ret;
+        }
+
+        /// <summary>
+        /// Equality comparer. Compares the two positions for equality within about 1cm.
+        /// </summary>
+        /// <param name="obj">The other position</param>
+        /// <returns>True if the two positions are almost identical</returns>
+        public override bool Equals(object? obj)
+        {
+            GeographicPosition? position = obj as GeographicPosition;
+
+            if (position == null)
+            {
+                return false;
+            }
+
+            if (((Math.Abs(position.Longitude - Longitude) < ComparisonEpsilon) &&
+                (Math.Abs(position.Latitude - Latitude) < ComparisonEpsilon)) &&
+                (Math.Abs(position.EllipsoidalHeight - EllipsoidalHeight) < ComparisonEpsilon))
+            {
+                return true;
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Equality comparer. Compares the two positions for equality within about 1cm.
+        /// </summary>
+        /// <param name="position">The other position</param>
+        /// <returns>True if the two positions are almost identical</returns>
+        public bool Equals(GeographicPosition? position)
+        {
+            if (position == null)
+            {
+                return false;
+            }
+
+            if (((Math.Abs(position.Longitude - Longitude) < ComparisonEpsilon) &&
+                 (Math.Abs(position.Latitude - Latitude) < ComparisonEpsilon)) &&
+                (Math.Abs(position.EllipsoidalHeight - EllipsoidalHeight) < ComparisonEpsilon))
+            {
+                return true;
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Returns a string representation of this position.
+        /// </summary>
+        /// <returns>A string representation in degrees, minutes and seconds for both latitude and longitude</returns>
+        // TODO: Add different formatting options and add parsing feature
```

Comment:
remove this, we'll add it when we need to, alternatively open the issue - still - TODOs should be out of the code as it can potentially stay there forever

Reply:
Been there, done that 😆 I'll remove it and write a ticket.